File: src/Entity/Menu.php
<?php

namespace App\Entity;

use App\Repository\MenuRepository;
use Doctrine\ORM\Mapping as ORM;
use Symfony\Component\Validator\Constraints as Assert;

/**
 * @ORM\Entity(repositoryClass=MenuRepository::class)
  * @ORM\Table(name="menu", indexes={@ORM\Index(name="fk_menu", columns={"refer_resto"})})

 */
class Menu
{ 

    /**
     * @ORM\GeneratedValue
     * @ORM\Id
     * @ORM\Column(type="integer")
     */
    private $codmenu;

    /**
     * @ORM\Column(type="string", length=255)
     * @Assert\NotBlank(message="descriptione is required")
     */
    private $libPlat;

    /**
     * @ORM\Column(type="float")
     * @Assert\NotBlank(message="prix is required")
     */
    private $prixPlat;

    /**
     * @ORM\Column(type="string", length=255)
 
     */
    private $referResto;
    /**
     * @ORM\ManyToOne(targetEntity=Categorie::class, inversedBy="Menu",cascade={"persist"})
     * @ORM\JoinColumn(name="categorie_id",referencedColumnName="id", onDelete="CASCADE")

 
     */
    private $Categorie;

      /**
     * @ORM\Column(type="string", length=50)
     * @Assert\NotBlank(message="nom plat is required")
     */
    private $NomPlat;

    public function __construct($referResto)
    {
        $this->referResto =$referResto;
 
    }
    

    public function getCodmenu(): ?int
    {
        return $this->codmenu;
    }

    public function setCodmenu(int $codmenu): self
    {
        $this->codmenu = $codmenu;

        return $this;
    }

    public function getLibPlat(): ?string
    {
        return $this->libPlat;
    }

    public function setLibPlat(string $libPlat): self
    {
        $this->libPlat = $libPlat;

        return $this;
    }

    public function getPrixPlat(): ?float
    {
        return $this->prixPlat;
    }

    public function setPrixPlat(float $prixPlat): self
    {
        $this->prixPlat = $prixPlat;

        return $this;
    }

    public function getReferResto(): ?string
    {
        return $this->referResto;
    }

    public function setReferResto(string $referResto): self
    {
        $this->referResto = $referResto;

        return $this;
    }
    
    public function getCategorie(): ?Categorie
    {
        return $this->Categorie;
    }

    public function setCategorie(?Categorie $Categorie): self
    {
        $this->Categorie = $Categorie;

        return $this;
    }

    public function getNomPlat(): ?string
    {
        return $this->NomPlat;
    }

    public function setNomPlat(string $nomPlat): self
    {
        $this->NomPlat = $nomPlat;

        return $this;
    }
}

File: src/trood/entity/Menu.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package trood.entity;

/**
 *
 * @author Rania
 */
public class Menu {
    private int codmenu;
    private String lib_plat;
    private double prix_plat;
    private String refer_resto;
    private String nom_plat;
    private String id_categorie;

    public Menu(int codmenu, String lib_plat, double prix_plat, String refer_resto, String nom_plat, String id_categorie) {
        this.codmenu = codmenu;
        this.lib_plat = lib_plat;
        this.prix_plat = prix_plat;
        this.refer_resto = refer_resto;
        this.nom_plat = nom_plat;
        this.id_categorie = id_categorie;
    }

 

    public int getCodmenu() {
        return codmenu;
    }

    public void setCodmenu(int codmenu) {
        this.codmenu = codmenu;
    }

    public String getLib_plat() {
        return lib_plat;
    }

    public void setLib_plat(String lib_plat) {
        this.lib_plat = lib_plat;
    }

    public double getPrix_plat() {
        return prix_plat;
    }

    public void setPrix_plat(double prix_plat) {
        this.prix_plat = prix_plat;
    }

    public String getRefer_resto() {
        return refer_resto;
    }

    public void setRefer_resto(String refer_resto) {
        this.refer_resto = refer_resto;
    }

    public String getNom_plat() {
        return nom_plat;
    }

    public void setNom_plat(String nom_plat) {
        this.nom_plat = nom_plat;
    }

    @Override
    public String toString() {
        return "Menu{" + "codmenu=" + codmenu + ", lib_plat=" + lib_plat + ", prix_plat=" + prix_plat + ", refer_resto=" + refer_resto + ", nom_plat=" + nom_plat + '}';
    }

    public Menu(String lib_plat, double prix_plat, String refer_resto, String nom_plat, String id_categorie) {
        this.lib_plat = lib_plat;
        this.prix_plat = prix_plat;
        this.refer_resto = refer_resto;
        this.nom_plat = nom_plat;
        this.id_categorie = id_categorie;
    }

    public String getId_categorie() {
        return id_categorie;
    }

    public void setId_categorie(String id_categorie) {
        this.id_categorie = id_categorie;
    }

 
    
    
    
}
